armi/nucDirectory/transmutations.py

```diff
@@ -212,7 +218,21 @@ class DecayMode(Transmutable):
 
     def __init__(self, parent, dataDict):
         Transmutable.__init__(self, parent, dataDict)
-        self.halfLifeInSeconds = dataDict["halfLifeInSeconds"]
+        self.halfLifeInSeconds = parent.halflife
+
+        # Check for user-defined value of half-life within the burn-chain data. If this is
+        # updated then prefer the user change and then note this to the user. Otherwise,
+        # maintain the default loaded from the nuclide bases.
+        userHalfLife = dataDict.get("halfLifeInSeconds", None)
+        if userHalfLife:
+            if userHalfLife != parent.halflife:
+                runLog.info(
+                    f"Half-life provided for {self} will be updated from "
+                    f"{parent.halflife:<15.11e} to {userHalfLife:<15.11e} seconds based on "
+                    f"user provided burn-chain data."
+                )
+
+                self.halfLifeInSeconds = dataDict["halfLifeInSeconds"]
```

Comment:
You don't have to do the dict lookup again here. You already have this value as `userHalfLife`.

Reply:
Fixed.